File: system-tests/gradle-multi-module/src/test/java/com/teamscale/tia/ConfigurationCacheTest.java
package com.teamscale.tia;

import com.teamscale.test.commons.SystemTestUtils;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Checks that the plugin is compatible with the <a
 * href="https://docs.gradle.org/current/userguide/configuration_cache.html">Gradle configuration cache</a>.
 */
public class ConfigurationCacheTest {

	@Test
	public void testConfigurationCache() throws Exception {
		var result = SystemTestUtils.runGradle("gradle-project", "clean", "tiaTests", "--configuration-cache");
		Assertions.assertThat(result.getStdout()).contains("BUILD SUCCESSFUL");
		result = SystemTestUtils.runGradle("gradle-project", "clean", "tiaTests", "--configuration-cache");
		Assertions.assertThat(result.getStdout()).contains("Reusing configuration cache");
	}
}

File: system-tests/gradle-multi-module/src/test/kotlin/com/teamscale/tia/ConfigurationCacheTest.kt
package com.teamscale.tia

import com.teamscale.test.commons.SystemTestUtils.runGradle
import org.assertj.core.api.Assertions
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

/**
 * Checks that the plugin is compatible with the [Gradle configuration cache](https://docs.gradle.org/current/userguide/configuration_cache.html).
 */
class ConfigurationCacheTest {
	/** Configuration cache is enabled via gradle.properties  */
	@Test
	@Throws(Exception::class)
	fun testConfigurationCache() {
		var result = runGradle("gradle-project", "clean", "systemTest")
		assertThat(result.stdout).contains("BUILD SUCCESSFUL")
		result = runGradle("gradle-project", "clean", "systemTest")
		assertThat(result.stdout).contains("Reusing configuration cache")
	}
}
